File: riceCoocker.php
<?php

class RiceCooker {
    private $riceType = '';
    private $waterLevel = 0;
    private $isCooking = false;

    public function setRiceType($type) {
        $this->riceType = $type;
        echo "Type de riz réglé sur : {$this->riceType}\n";
    }

    public function setWaterLevel($level) {
        if ($level < 0 || $level > 100) {
            throw new Exception('Niveau d\'eau invalide. Veuillez choisir une valeur entre 0 et 100.');
        }

        $this->waterLevel = $level;
        echo "Niveau d'eau réglé sur : {$this->waterLevel}%\n";
    }

    public function startCooking() {
        if (empty($this->riceType)) {
            throw new Exception('Veuillez sélectionner un type de riz avant de commencer la cuisson.');
        }

        if ($this->waterLevel == 0) {
            throw new Exception('Veuillez définir le niveau d\'eau avant de commencer la cuisson.');
        }

        echo "Cuisson en cours...\n";
        $this->isCooking = true;

        // Simulation de la cuisson - remplacez cela par une logique de détection réelle
        sleep(5); // Cuisson pendant 5 secondes (à remplacer par une logique réelle)

        echo "La cuisson est terminée.\n";
        $this->isCooking = false;
    }

    public function checkCookingStatus() {
        echo "État du cuiseur à riz - Type de riz: {$this->riceType}, Niveau d'eau: {$this->waterLevel}%, Cuisson en cours: " . ($this->isCooking ? "Oui" : "Non") . "\n";
    }
}

// Utilisation du cuiseur à riz
$riceCooker = new RiceCooker();

try {
    $riceCooker->setRiceType('riz blanc');
    $riceCooker->setWaterLevel(50);
    $riceCooker->startCooking();
    $riceCooker->checkCookingStatus();
} catch (Exception $error) {
    echo "Erreur: {$error->getMessage()}\n";
}
?>

File: riceCoocker.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Structure représentant le cuiseur à riz
typedef struct {
    char riceType[50];
    int waterLevel;
    int isCooking;
} RiceCooker;

// Fonction pour régler le type de riz
void setRiceType(RiceCooker *cooker, const char *type) {
    strcpy(cooker->riceType, type);
    printf("Type de riz réglé sur : %s\n", cooker->riceType);
}

// Fonction pour régler le niveau d'eau
void setWaterLevel(RiceCooker *cooker, int level) {
    if (level < 0 || level > 100) {
        fprintf(stderr, "Niveau d'eau invalide. Veuillez choisir une valeur entre 0 et 100.\n");
        exit(EXIT_FAILURE);
    }

    cooker->waterLevel = level;
    printf("Niveau d'eau réglé sur : %d%%\n", cooker->waterLevel);
}

// Fonction pour démarrer la cuisson
void startCooking(RiceCooker *cooker) {
    if (strlen(cooker->riceType) == 0) {
        fprintf(stderr, "Veuillez sélectionner un type de riz avant de commencer la cuisson.\n");
        exit(EXIT_FAILURE);
    }

    if (cooker->waterLevel == 0) {
        fprintf(stderr, "Veuillez définir le niveau d'eau avant de commencer la cuisson.\n");
        exit(EXIT_FAILURE);
    }

    printf("Cuisson en cours...\n");
    cooker->isCooking = 1;

    // Simulation de la cuisson - remplacez cela par une logique de détection réelle
    for (int i = 0; i < 5; i++) {
        // Temporisation d'une seconde
        sleep(1);
    }

    printf("La cuisson est terminée.\n");
    cooker->isCooking = 0;
}

// Fonction pour vérifier l'état de la cuisson
void checkCookingStatus(const RiceCooker *cooker) {
    printf("État du cuiseur à riz - Type de riz: %s, Niveau d'eau: %d%%, Cuisson en cours: %s\n",
           cooker->riceType, cooker->waterLevel, cooker->isCooking ? "Oui" : "Non");
}

int main() {
    // Utilisation du cuiseur à riz
    RiceCooker riceCooker;
    memset(&riceCooker, 0, sizeof(RiceCooker));

    try {
        setRiceType(&riceCooker, "riz blanc");
        setWaterLevel(&riceCooker, 50);
        startCooking(&riceCooker);
        checkCookingStatus(&riceCooker);
    } catch (const char *error) {
        fprintf(stderr, "Erreur: %s\n", error);
        return EXIT_FAILURE;
    }

    return EXIT_SUCCESS;
}
